### PR TITLE
Test case to perform orchestrator pause and resume operations

### PR DESCRIPTION
**Pull request by manasagowri**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Review by psathyan — COMMENTED** [review]
Thanks for splitting.

File: ceph/ceph_admin/orch.py [hunk 1]
```diff
@@ -237,3 +241,53 @@ def op(self, op, config):
...
+        cmd.append("status")
+
+        if config and config.get("args"):
+            args = config.get("args")
+            if args["detail"]:
+                cmd.append("--detail")
+
+        return self.shell(args=cmd)
+
+    def verify_pause_status(self, op) -> None:
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply ceph/ceph_admin/orch.py:282]: Done

File: ceph/ceph_admin/orch.py [hunk 2]
```diff
@@ -237,3 +241,53 @@ def op(self, op, config):
...
+        """Verify if the orchestrator operations are paused or resumed
+
+        Args:
+            op (str): pause/resume based on whether the pause or resume status to be checked
+        """
+        config = {"command": "status", "base_cmd_args": {"format": "json"}}
+        out, _ = self.orch_status(config)
+        if op == "pause" and loads(out)["paused"]:
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply ceph/ceph_admin/orch.py:290]: Done.

File: ceph/ceph_admin/orch.py [hunk 3]
```diff
@@ -237,3 +241,53 @@ def op(self, op, config):
         base_cmd.extend(config.get("pos_args"))
 
         return self.shell(args=base_cmd)
+
+    def orch_status(self, config) -> bool:
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply ceph/ceph_admin/orch.py:245]: No, rename done.

File: suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml [hunk 4]
```diff
@@ -277,11 +280,27 @@ tests:
...
-
-
-
-
-
+  - test:
+      name: Pause orchestrator background operations using cephadm
+      desc: Apply ceph orch pause to pause orchestrator background operations
+      module: test_orch.py
+      polarion-id:
```
Comment (psathyan) [inline suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml:287]: Please update the Polarion IDs

**Review by psathyan — COMMENTED** [review]
Looks good to me... kindly update the Polarion IDs

File: suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml (hunk 4 above)
Comment (manasagowri) [reply suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml]: Done.

**Review by psathyan — APPROVED** [review]

**Pull request merged**